diff --git a/DEPS b/DEPS
index 102f46264b..9b9cb5ae9b 100644
--- a/DEPS
+++ b/DEPS
@@ -270,7 +270,7 @@ deps = {
       'dep_type': 'cipd',
   },
   'third_party/requests': {
-      'url': Var('chromium_url') + '/external/github.com/kennethreitz/requests.git' + '@' + '2c2138e811487b13020eb331482fb991fd399d4e',
+      'url': Var('chromium_url') + '/external/github.com/kennethreitz/requests.git',
       'condition': 'checkout_android',
   },
   'third_party/zlib':
